app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -1980,6 +2034,18 @@ class BrowserTabViewModel(
         return fireproofWebsites?.any { it.domain == domain } ?: false
     }
 
+    private fun isBookmark(url: String?): Boolean {
+        if (url == null) return false
+        val bookmarks = bookmarksState.value
+        return bookmarks?.any { it.url == url } ?: false
+    }
+
+    private fun isFavorite(url: String?): Boolean {
+        if (url == null) return false
+        val favorites = currentCtaViewState().favorites
```

Comment:
I'm a bit worried about using this ViewState for our goal. Right now is ok, but imagine that at some point we change the logic that updates the CtaViewState (e.g: we want to show only 10 favorites in the home tab), that will break our logic.

This scenario is different from how we are dealing with bookmarks. For bookmarks, we are using directly an observer. 

I have some proposals:
1. instead of using an attribute from this class, always go through the repository (similar to how we update `isWhitelisted` in the same  viewstate)
2. similar approach to bookmarks.

Let's sync on zoom about this :)